### PR TITLE
dua: init at 2.1.5

### PR DESCRIPTION
**Pull request by killercup**
> A tool to conveniently learn about the disk usage of directories, fast!

Adds lock file that was missing from upstream repo.

Adds myself to maintainers list.

###### Motivation for this change

Add initial version of nice alternative to `du`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/dua/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ stdenv, lib, darwin
+, rustPlatform, fetchFromGitHub
+, openssl, pkgconfig }:
```
Comment (aanderse) [inline pkgs/tools/misc/dua/default.nix]: Please remove unused arguments.

File: pkgs/tools/misc/dua/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Byron";
+    repo = "dua-cli";
+    rev = "v${version}";
+    sha256 = "0m6a6pv998wzm9bpc8gykw4vvirjwgic48df7pawj299b3h9xg6d";
+  };
+
+  cargoSha256 = "18r9b41ir7y4pdznk5w93923c1pagkw5g2gh7206k0zyv2119j29";
+  cargoPatches = [ ./cargo-lock.patch ];
```
Comment (aanderse) [inline pkgs/tools/misc/dua/default.nix]: This patch belongs upstream, does it not?

File: pkgs/tools/misc/dua/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  meta = with lib; {
+    description = "A tool to conveniently learn about the disk usage of directories, fast!";
+    homepage = https://github.com/Byron/dua-cli;
```
Comment (aanderse) [inline pkgs/tools/misc/dua/default.nix]: Please quote.

File: pkgs/tools/misc/dua/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ killercup ];
```
Comment (aanderse) [inline pkgs/tools/misc/dua/default.nix]: Please create a separate commit adding yourself to the maintainers list file.

**Review by aanderse — CHANGES_REQUESTED** [review]
Looking good! Just a few changes I've mentioned which I hope you'll find helpful. Thanks for contributing this tool, I'm very interested in trying it.

File: pkgs/tools/misc/dua/default.nix (hunk 2 above)
Comment (killercup) [reply pkgs/tools/misc/dua/default.nix]: I upstreamed it in https://github.com/Byron/dua-cli/pull/19 but there has not been a release containing it yet

Comment (aanderse) [reply pkgs/tools/misc/dua/default.nix]: That's great! You're on top of it :smile:

**Comment by killercup** [comment]
Thanks for the review, @aanderse! Updated to freshly released 2.1.5 and addressed all comments :)

File: pkgs/tools/misc/dua/default.nix (hunk 1 above)
Comment (aanderse) [reply pkgs/tools/misc/dua/default.nix]: @killercup It looks like you don't need `stdenv`.

Comment (killercup) [reply pkgs/tools/misc/dua/default.nix]: Ah sorry, missed that one!

**Comment by aanderse** [comment]
@GrahamcOfBorg build dua

**Comment by aanderse** [comment]
After reading upstream documentation I'm confused as to what the package name is: `dua` or `dua-cli`. I think it is `dua-cli`.

As per the manual:
> The name attribute should be identical to the upstream package name.

**Comment by killercup** [comment]
@aanderse hm, I assumed this is only called dua-cli because it would otherwise conflict on crates.io with another package called dua. The binary is also called dua and it only ever referred to as dua except for the repo/rust library name.

It's up to you. I don't mind changing it.

**Review by aanderse — APPROVED** [review]
Merging as is because what you said makes sense to me. Thank you very much for your contribution @killercup!

**Comment by killercup** [comment]
Awesome, thanks again for the great feedback, @aanderse!

**Pull request merged**